Mathlib/Order/SuccPred/Inverse.lean

```diff
@@ -0,0 +1,344 @@
+/-
+Copyright (c) 2024 Violeta Hernández Palacios. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Violeta Hernández Palacios
+-/
+import Mathlib.Order.SuccPred.Basic
+
+/-!
+# "Predecessor" from successor
+
+For a `SuccOrder`, we define `pred' a` as some non-maximal element `b` with `succ b = a` if it
+exists, and `pred' a = a` otherwise. In a successor-predecessor partial order, `pred'` and `pred`
+coincide, but in general, `pred'` need not form a valid `PredOrder`.
+
+`pred'` is most nicely behaved in a linear order with no maximum, where it forms a Galois
+connection with `succ`.
+
+We also dually define `succ'` for a `PredOrder`.
+-/
+
+namespace Order
+
+open Set OrderDual
+
+variable {α : Type*}
+
+/-! ### Predecessor from successor -/
+
+
+open Classical in
+/-- A predecessor-like operation defined in a succesor order.
+
+If `a` is the succesor of some non-maximal `b`, `pred' a` returns an arbitrary such `b`. Otherwise,
+`pred' a = a`. -/
+noncomputable def pred' [Preorder α] [SuccOrder α] (a : α) : α :=
+  if h : ∃ b, ¬ IsMax b ∧ succ b = a then Classical.choose h else a
```

Comment:
If I'm not mistaken this condition is equivalent to `∃ b < a, succ b = a`, and I think it might be better—the `IsMax` here feels a bit weird.